Src/Support/Google.Apis.Core/Http/HttpClientFactory.cs

```diff
@@ -23,139 +23,22 @@ namespace Google.Apis.Http
-        /// It's unlikely, but if you need the underlying http client handler to follow redirects itself
-        /// set this to false.
-        /// </param>
-        /// <param name="httpMessageHandlerFactory">
-        /// Factory for obtaining the underlying <see cref="HttpMessageHandler"/>
-        /// of the <see cref="ConfigurableHttpClient"/> returned by
-        /// <see cref="IHttpClientFactory.CreateHttpClient(CreateHttpClientArgs)"/>.
-        /// May be null in which case <see cref="CreateClientHandler"/> will be used.
-        /// You may set this value, for instance, if you are using System.Net.Http.IHttpMessageHandlerFactory
-        /// to obtain <see cref="HttpMessageHandler"/> instances. You can read here to get more information on why
-        /// this might be necessary:
-        /// https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests
-        /// </param>
-        /// <param name="underlyingHttpClientHandlerSelector">
-        /// Used to extract the underlying <see cref="HttpClientHandler"/> from the given
-        /// <see cref="HttpMessageHandler"/>. The <see cref="HttpClientHandler"/> is used
-        /// to configure redirects and automatic decrompression.
-        /// May be null in which case a casting of the given <see cref="HttpMessageHandler"/> to
-        /// <see cref="HttpClientHandler"/> will be attempted.
-        /// This function will only be called on instances returned by <paramref name="httpMessageHandlerFactory"/>.
-        /// This function may return null in which case redirects and automatic decompression
-        /// will be configured in the same way as they would be for an <see cref="HttpClientHandler"/>
-        /// that doesn't support redirect configuration and automatic decompression.
-        /// </param>
-        public HttpClientFactory( // TODO: Consider creating an Initializer to avoid breaking changes in the future.
-            bool disposeInternalHandler, bool disableInternalHandlerAutoredirect,
-            Func<bool, HttpMessageHandler> httpMessageHandlerFactory, Func<HttpMessageHandler, HttpClientHandler> underlyingHttpClientHandlerSelector)
-        {
-            DisposeInternalHandler = disposeInternalHandler;
-            DisableInternalHandlerAutoRedirect = disableInternalHandlerAutoredirect;
-            // We can ignore mayCompress when using CreateClientHandlers, because it's clear on the docs
-            // that this method shouldn't try to configure automatic compression.
-            HttpMessageHandlerFactory = httpMessageHandlerFactory ?? ( mayCompress => CreateClientHandler());
-            _handlerExternalFactory = httpMessageHandlerFactory != null;
-            UnderlyingHttpClientHandlerSelector = underlyingHttpClientHandlerSelector ?? (mh => mh as HttpClientHandler);
-        }
+        /// <summary>The class logger.</summary>
```

Comment:
Just as a reminder for anyone reviewing. These changes in the second commit are just undoing the the changes in the first commit. In this PR commit 1 and commit 2 are alternative implementations of the same thing.